File: store/src/main/java/com/min/store/product/repository/ProductRepository.java
package com.min.store.product.repository;

import com.min.store.product.entity.ProductEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ProductRepository extends JpaRepository<ProductEntity, Long> {
    Optional<ProductEntity> findByName(String name);
    boolean existsByName(String name);
}

File: store/store-product-api/src/main/kotlin/com/store/product/api/repository/ProductRepository.kt
package com.store.product.api.repository

import com.store.product.api.entity.ProductEntity
import org.springframework.data.jpa.repository.JpaRepository

interface ProductRepository : JpaRepository<ProductEntity, Long> {
    fun findByName(name: String): ProductEntity?
    fun existsByName(name: String): Boolean
}
